src/pages/index.js

```diff
@@ -0,0 +1,393 @@
+  console.log(userInfo);
+
+  avatarImageSrc.src = userInfo.avatar;
+
+})
+  .catch((err) => {
+    console.error(err);
+  });
+
+//Profile elements
+const profileEditbutton = document.querySelector('.profile__edit-btn');
+const profileAddbutton = document.querySelector('.profile__add-btn');
+const avatarModalBtn = document.querySelector('.profile__avatar-btn');
+const profileName = document.querySelector('.profile__name');
+const profileDescription = document.querySelector('.profile__description');
+
+// Form elements
+const editModal = document.querySelector('#edit-modal');
+const editFormElement = editModal.querySelector('.modal__form');
+const editModalCloseBtn = editModal.querySelector('.modal__close-btn');
+const editModalNameInput = editModal.querySelector('#profile-name-input');
+const editModalDescriptionInput = editModal.querySelector(
+  '#profile-description-input'
+);
+
+// Card Modal elements
+const cardModal = document.querySelector('#add-card-modal');
+const cardForm = cardModal.querySelector('.modal__form');
+const cardSubmitBtn = cardModal.querySelector('.modal__submit-btn');
+const cardNameInput = cardModal.querySelector('#add-card-name-input');
+const cardLinkInput = cardModal.querySelector('#add-card-link-input');
+const cardModalCloseBtn = cardModal.querySelector('.modal__close-btn');
+
+// Preview Modal elements
+const previewModal = document.querySelector('#preview-modal');
+const previewModalImageEl = previewModal.querySelector('.modal__image');
+const previewModalCaptionEl = previewModal.querySelector('.modal__caption');
+const previewModalCloseBtn = previewModal.querySelector('.modal__close-btn');
+
+//Card related elements
+const cardTemplate = document.querySelector('#card-template');
+const cardsList = document.querySelector('.cards__list');
+
+// select the avatar modal and its elements
+const avatarModal = document.querySelector('#avatar-modal');
+const avatarForm = avatarModal.querySelector('.modal__form');
+const avatarSubmitBtn = avatarModal.querySelector('.modal__submit-btn');
+const avatarModalCloseBtn = avatarModal.querySelector('.modal__close-btn');
+const avatarInput = avatarModal.querySelector('#profile-avatar-input');
+
+// Delete Modal elements
+const deleteModal = document.querySelector('#delete-modal');
+const deleteForm = deleteModal.querySelector('.modal__form');
+const deleteModalCancelBtn = deleteModal.querySelector('#delete-cancel-btn');
+let selectedCard;
+let selectedCardId;
+
+function openModal(modal) {
+  modal.classList.add('modal_opened');
+  document.addEventListener('keydown', handleEscapeKey);
+}
+
+function closeModal(modal) {
+  modal.classList.remove('modal_opened');
+  document.removeEventListener('keydown', handleEscapeKey);
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  // editModalSubmitBtn.textContent = 'Saving...';
+  // change text content to "Saving..."
+  const editModalSubmitBtn = evt.submitter;
+  editModalSubmitBtn.textContent = 'Saving...';
```

Comment:
` editModalSubmitBtn.textContent = 'Saving...';` is not needed here because it's done inside `setButtonText` in line 148

Reply:
* Some of my remarks haven't been corrected, starting from here https://github.com/Jellyfish15/se_project_spots/pull/1#discussion_r2119551259